src/nsntrace.c

```diff
@@ -245,7 +253,9 @@ netns_main(void *arg) {
 	} else if (child_pid > 0) { /* parent - tracer */
 		struct timespec timeout = { 0 };
 
-		waitpid(child_pid, NULL, 0);
+		waitpid(child_pid, &status, 0);
+		ret = WIFEXITED(status) ? WEXITSTATUS(status) : EXIT_FAILURE;
```

Comment:
While the trinary operator is concise I actually prefer this as an if-statement. To make it easier to read.

Something like:
if (WIFEXITED(status))
    ret = ...
else
   ret = ...